RemoteMetaMaskProvider.js

```diff
@@ -35,60 +50,78 @@ class RemoteMetaMaskProvider {
     if (Object.prototype.hasOwnProperty.call(translateMethod, method)) {
       return translateMethod[method];
     }
+
     return method;
   }
-  _guid() {
-    this.s4 = () =>
-      Math.floor((1 + Math.random()) * 0x10000)
-        .toString(16)
-        .substring(1);
-    return `
-      ${this.s4()}
-      ${this.s4()}
-      -
-      ${this.s4()}
-      -
-      ${this.s4()}
-      -
-      ${this.s4()}
-      -
-      ${this.s4()}
-      ${this.s4()}
-      ${this.s4()}
-    `.replace(/[\n\r]+ */g, '');
+
+  // When connected to a remote network, the return values for "gasPrice" and
+  // "value" are strings, so we will need to properly format them for ethers.
+  // Ideally we would use the big number type from bn.js or bignumber.js but
+  // ethers does not support any big number type other than it's own.
+  static formatResult(_result) {
+    const result = _result;
+
+    // Format "gasPrice"
+    if (result && typeof result.gasPrice === 'string') {
+      result.gasPrice = Number(result.gasPrice);
+    }
+
+    // Format "value"
+    if (result && typeof result.value === 'string') {
+      result.value = Number(result.value);
+    }
+
+    return result;
   }
+
+  // Define send method
   send(_payload, _callback) {
     if (!this._connector.ready()) {
       return _callback(
-        new Error("Can't send. Not connected to a MetaMask socket."),
+        new Error('Unable to send. Not connected to a MetaMask socket.'),
       );
     }
-    // Because requests are handled across a WebSocket they need to be
-    // associated with their callback with an ID which is returned with the
-    // response.
-    const requestId = this._guid();
-    const payload = _payload;
+
+    // Because requests are handled across a WebSocket, their callbacks need to
+    // be associated with an ID which is returned with the response.
+    const requestId = RemoteMetaMaskProvider.generateRequestId();
```

Comment:
We like to use `this.constructor.xxx` instead.

Reply:
👍